cumulusci/robotframework/locator_manager.py

```diff
@@ -0,0 +1,149 @@
+from robot.libraries.BuiltIn import BuiltIn
+import functools
+from cumulusci.core.utils import dictmerge
+
+"""
+This module supports managing multiple location strategies. It
+works like this:
+
+0. Locators are stored in a global LOCATORS dictionary. The keys
+   are the locator prefixes, the values are dictionaries containing
+   the locators
+
+1. Libraries can register a dictionary of locators with a prefix
+   (eg: register_locators("npsp", {...})). These get added to
+   LOCATORS
+
+2. Open Test Brower calls selenium's add_location_strategy for each
+   registered set of locators. (Note: This cannot be called before a
+   browser is open.
+
+3. Keywords can use dot notation to refer to locators. A colon
+   separates the prefix from the locator, and a locator from a
+   comma-separated string of arguments
+
+Example:
+
+    from a keyword library:
+
+    | from cumulusci.robotframework.locator_manager import register_locators
+    |
+    | register_locators("example", {"dialog": {"button": "//button[text()='{}']"}})
+
+    in a test:
+
+    | page should contain element  example:dialog.button:Save
+
+    ... will result in "//button[text()='Save']" being passed
+    to 'page should contain element'
+
+"""
+
+LOCATORS = {}
+
+
+def register_locators(prefix, locators):
+    """Register a strategy with a set of locators or a keyword
+
+    If the prefix is already known, merge in the new locators.
+    """
+    if prefix in LOCATORS:
+        BuiltIn().log(f"merging keywords for prefix {prefix}", "DEBUG")
+        dictmerge(LOCATORS[prefix], locators)
+    else:
+        BuiltIn().log(f"registering keywords for prefix {prefix}", "DEBUG")
+        LOCATORS[prefix] = locators
+
+
+def add_location_strategies():
+    """Call selenium's add_location_strategy keyword for all strategies"""
+
+    # selenium throws an error if the location strategy already
+    # exists, so we use a flag to make sure this code is called
+    # only once.
+    selenium = BuiltIn().get_library_instance("SeleniumLibrary")
+    for (prefix, strategy) in LOCATORS.items():
+        try:
+            BuiltIn().log(f"adding location strategy for '{prefix}'", "DEBUG")
+            if isinstance(strategy, dict):
+
+                def strategy(parent, locator, tag, constraints):
+                    locate_element(prefix, parent, locator, tag, constraints)
```

Comment:
This strategy function is no longer used since you switched to functools.partial (thanks coverage)

Reply:
Hah! Good catch. I saw the coverage report and thought it was wrong, because I "knew" the code was working. I had forgotten I switched to `functools.partial` instead of using the embedded function. Now I'm up to 100% coverage of the new code. w00t!